service/builder/exporters_builder.go

```diff
@@ -168,8 +168,11 @@ func (eb *ExportersBuilder) buildExporter(
 
 	inputDataTypes := exportersInputDataTypes[config]
 	if inputDataTypes == nil {
-		// No data types where requested for this exporter. This can only happen
-		// if there are no pipelines associated with the exporter.
+		if config.IsEnabled() {
+			return nil, fmt.Errorf("Exporter " + config.Name() +
```

Comment:
Unclear why this should be an error. It seems that giving a warning (below) should be sufficient.

Reply:
As I mentioned earlier, the created eb will have nil stopFunc, which causes panic during process exiting.

Reply:
@huikang:

I do not see why the panic during shutdown is conflated with the proposed behavior change during configuration loading at startup.

There is a piece of code which panics when it should not. This needs to be fixed without changing unrelated functionality: logging of warning on certain configuration.

If you think that it is better to give an error in this situation instead of giving a warning then that is  best to be discussed separately in a separate issue. For now I do not see good reasons for changing a warning to an error. If you can bring such reasons I will be happy to consider.

Otherwise, I think it is best to fix the panic bug during shutdown without changing the intended behavior of the program during configuration reading, i.e. keep the warning.

In this case the root cause of the panic is that `buildExporter()` function violates the contract of `builtExporter` type. It is expected that every valid `builtExporter` instance must have non-nil `stop` member.

The correct solution is to fix `buildExporter()` function to honour the type contract and supply non-nil `stop` member. This can be done by adding one line at line 175 before return statement:
```go
exporter.stop = func() error { return nil }
```
This should fix the panic bug without changing intended application behavior during startup.

Reply:
@tigrannajaryan , yours is better. My concern was about the behavior of the stop function. Now it seems a simple `func() error { return nil }` should be good enough.

I will update the PR.

Reply:
@huikang if you can also add a test that verifies this edge case it would be great.

Reply:
Ah, I see, you already did. Thanks!